robot-core/src/main/java/org/obolibrary/robot/checks/Report.java

```diff
@@ -0,0 +1,204 @@
+package org.obolibrary.robot.checks;
+
+import java.net.URL;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import org.obolibrary.robot.OntologyHelper;
+import org.obolibrary.robot.UnmergeOperation;
+import org.semanticweb.owlapi.model.IRI;
+import org.semanticweb.owlapi.model.OWLOntology;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class Report {
+
+  /** Logger. */
+  private static final Logger logger = LoggerFactory.getLogger(UnmergeOperation.class);
+
+  /** Line separator */
+  private static final String newLine = System.getProperty("line.separator");
+
+  /** Reporting levels. */
+  private static final String INFO = "INFO";
+
+  private static final String WARN = "WARN";
+  private static final String ERROR = "ERROR";
+
+  /** Maps of rules and the violations for each level */
+  public Map<String, List<Violation>> info;
+
+  public Map<String, List<Violation>> warn;
+  public Map<String, List<Violation>> error;
+
+  /** Counts of violations for each level */
+  private Integer infoCount;
+
+  private Integer warnCount;
+  private Integer errorCount;
+
+  /** Labels from the ontology */
+  private Map<IRI, String> labels;
+
+  /**
+   * Given an ontology, create a new report on it.
+   *
+   * @param ontology OWLOntology to report on
+   */
+  public Report(OWLOntology ontology) {
+    info = new HashMap<>();
+    warn = new HashMap<>();
+    error = new HashMap<>();
+
+    infoCount = 0;
+    warnCount = 0;
+    errorCount = 0;
+
+    // get ontology-specific labels
+    labels = OntologyHelper.getLabels(ontology);
```

Comment:
This is brutally slow, even for a smallish ontology such as OBI. My code, my fault.

SPARQL would be faster. The most efficient approach would be to get only the labels that we need, once the violations have been collected.